libminifi/src/utils/net/DNS.cpp

```diff
@@ -127,4 +69,9 @@ nonstd::expected<std::string, std::error_code> reverseDnsLookup(const asio::ip::
   return results->host_name();
 }
 
+std::string getMyHostName() {
+  static const std::string HOSTNAME = asio::ip::host_name();
+  return HOSTNAME;
+}
```

Comment:
Is this static caching actually needed? Is querying the hostname too expensive?

Reply:
I'm not sure how expensive is it, but as it should not change at all, I think caching it wouldn't hurt.

Reply:
Hostname can change at runtime. On Linux, you can use the `hostname` command.